### ISSUE TITLE
store photo dimensions

### ISSUE THREAD

**Issue by kueda** [inaturalist/inaturalist#682]
A lot of the more interesting visualizations and layouts we could do with photos require knowing their dimensions a priori. This data comes from almost every photo API and we can derive it from all photos uploaded directly to iNat, so we certainly *can* store this stuff, we're just not.

http://photoswipe.com is what got me thinking about this which seems like it might want dimensions for every image URL, which changes the requirements a bit, but still seems doable. They also linked to https://github.com/sdsykes/fastimage, which seems like it would be a useful tool for adding these values to existing photos.

**Update**: we decided to store dimensions for all image sizes.

**Comment by pleary** [comment]
I think #702 contains most or all of what we need. It will start storing an additional field in photo.metadata which contains a hash of all the sizes of all photo styles.

I also made a rake task to fetch dimensions for any photos that don't yet have them. It uses https://github.com/sdsykes/fastimage which I found to be pretty fast effective for minimizing bandwidth. It will only download the original photo and extrapolate the dimensions of the remaining photo styles, so we don't have to make (photo * 6) HTTP requests.
